checker.cpp

```diff
@@ -1,10 +1,70 @@
 #include <iostream>
 #include <assert.h>
 #include <unistd.h>
+#include <string>
+#include <unordered_map>
 using namespace std;
 
+string getPulseRateCondition(float pulseRate)
+{
+    unordered_map<string, pair<float, float>> conditions = { {"BRADYCARDIA", {0,59.99}}, 
+                                                             {"NEAR BRADYCARDIA", {60, 61.59}}, 
+                                                             {"PULSERATE NORMAL", {61.60,98.49}},
+                                                             {"NEAR TACHYCARDIA", {98.50,100}},
+                                                             {"TACHYCARDIA",{100,300}}
+                                                            };
+    string pulseCondition;
+    for(auto i: conditions)
+    {
+        if(pulseRate >= i.second.first && pulseRate <= i.second.second)
+        {
+            pulseCondition = i.first;
+            break;
+        }
+    }
+    return pulseCondition;
+}
+
+string getSpo2Condition(float spo2)
+{
+    unordered_map<string, pair<float,float>> conditions = { {"HYPOXEMIA", {0,89.99}},
+                                                            {"NEAR HYPOXEMIA", {90,91.35}},
+                                                            {"Spo2 NORMAL", {91.36,100}},
+                                                            };
+    string spo2Condition;
+    for(auto i: conditions)
+    {
+        if(spo2 >= i.second.first && spo2 <= i.second.second)
```

Comment:
This code-pattern looks very similar to the previous function. Can you extract it to a common place? It will make easier to add common functionality - e.g, error-handling when the measure is more than the max